src/test/java/io/vertx/test/codegen/testapi/jsonmapper/MyEnumWithCustomConstructor.java

```diff
@@ -0,0 +1,40 @@
+package io.vertx.test.codegen.testapi.jsonmapper;
+
+import io.vertx.codegen.annotations.VertxGen;
+
+/**MyEnumWithCustomConstructor doc*/
+@VertxGen
+public enum MyEnumWithCustomConstructor {
```

Comment:
it should be renamed with `XXXWithCustomFactory` because _constructor_ has a very specific meaning

Reply:
ok, I did the required changes... :D

But it is really called an enum constructor, not a factory :)
Look here: https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html